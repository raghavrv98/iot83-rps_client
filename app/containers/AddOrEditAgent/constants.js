/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
