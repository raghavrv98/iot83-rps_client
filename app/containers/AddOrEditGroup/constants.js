/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
