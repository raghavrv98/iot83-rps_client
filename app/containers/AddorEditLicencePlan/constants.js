/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";