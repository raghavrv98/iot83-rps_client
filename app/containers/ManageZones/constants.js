/*
 *
 * ManageZones constants
 *
 */

export const DEFAULT_ACTION = "app/ManageZones/DEFAULT_ACTION";

export const GET_ZONE_LIST ="app/ManageZones/GET_ZONE_LIST"
export const GET_ZONE_LIST_SUCCESS ="app/ManageZones/GET_ZONE_LIST_SUCCESS"
export const GET_ZONE_LIST_FAILURE ="app/ManageZones/GET_ZONE_LIST_FAILURE"

export const ZONE_DELETE = "app/ManageZones/ZONE_DELETE"
export const ZONE_DELETE_SUCCESS = "app/ManageZones/ZONE_DELETE_SUCCESS"
export const ZONE_DELETE_FAILURE = "app/ManageZones/ZONE_DELETE_FAILURE"
