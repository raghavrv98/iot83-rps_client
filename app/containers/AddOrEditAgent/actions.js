/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
