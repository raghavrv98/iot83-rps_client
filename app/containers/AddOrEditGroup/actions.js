/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
