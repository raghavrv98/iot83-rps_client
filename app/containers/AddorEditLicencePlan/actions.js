/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}