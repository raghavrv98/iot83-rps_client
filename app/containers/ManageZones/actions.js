/*
 *
 * ManageZones actions
 *
 */

import { GET_ZONE_LIST ,ZONE_DELETE } from "./constants";

export function getZoneList(plantId, pipelineId){
  return{
    type:GET_ZONE_LIST,
    plantId, pipelineId
  }
}
export function zoneDeleteHandler(plantId, pipelineId, multiple, zoneId){
  return{
    type:ZONE_DELETE,
    plantId, pipelineId, zoneId, multiple
  }
}