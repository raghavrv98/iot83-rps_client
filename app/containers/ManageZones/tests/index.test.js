import React, { Component } from 'react';
import { shallow, mount} from 'enzyme';
import history from "utils/history";
import { getZoneList, zoneDeleteHandler } from "../actions";
import { GET_ZONE_LIST, ZONE_DELETE } from "../constants";
import { ManageZones, mapDispatchToProps } from "../index";
const zoneList = {result:[
    {
      _id: "9bed0d5ca79011e9863a0242ac110016",
      plantId: "9bed0d5ca79011e9863a0242ac110016",
      result: "demo",
      pipelineName: "demo",
      plantName: "demo",
      CircuitId: {},
      DesignPF: {},
      DesignVolts: {},
      EndChain: {},
      FeedChain: {},
      MaintainAmps: {},
      MaintainTemp: {},
      PanelId: {},
      StartupAmps: {},
      StartupTemp: {}
    }
  ]};

describe('manageAgentsSaga Saga', () => {

    const match = {
        params: {
            plantId: "demo",
            pipelineId: "demo"
        }
    };

    it('Expect to have unit tests specified', () => {
        expect(true).toEqual(true);
    });

    it('match snapshots', () => {
        const wrap = shallow(<ManageZones match={match} getZoneList={getZoneList} history={history}/>)
        expect(wrap).toMatchSnapshot()
    })

    it("test componentWillMount", async () => {
        const fetchZonesList = jest.fn();
        spyOn(ManageZones.prototype, "componentWillMount").and.callThrough();
        shallow(
            <ManageZones
                history={history}
                match={match}
                getZoneList={getZoneList}
            />
        );
        expect(ManageZones.prototype.componentWillMount).toHaveBeenCalledTimes(1);
        expect(fetchZonesList).toHaveBeenCalledTimes(0);
    });

    it('should test component when render', () => {
        spyOn(ManageZones.prototype, 'render').and.callThrough();
        const wrapper = shallow(<ManageZones getZoneList={getZoneList} zonesList={undefined} zonesListFailure={undefined} getzonesDelete={undefined} getzoneDeleteFailure={undefined} match={match} />);
        expect(wrapper).toBeDefined();
        expect(ManageZones.prototype.render).toHaveBeenCalledTimes(1);
    })

    it('should test component when recieve props', () => {
        spyOn(ManageZones.prototype, 'componentWillReceiveProps').and.callThrough();
        const wrapper = shallow(<ManageZones getZoneList={getZoneList} match={match} zonesList={undefined} zonesListFailure={undefined} getzonesDelete={undefined} getzoneDeleteFailure={undefined} />);
        expect(wrapper).toBeDefined();
        wrapper.setState({ fetchZonesList:zoneList.result, pipelineName: zoneList.result[0].pipelineName, plantName: zoneList.result[0].plantName });
        wrapper.setProps({ zonesList: zoneList, zonesListFailure: "No Error Found.", getzonesDelete: "9bed0d5ca79011e9863a0242ac110016", getzoneDeleteFailure: "No Error Found in delete." });
        expect(ManageZones.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
    })

    it("clicking add button trigger onClick event", async () => {
        let expectedCount = history.length + 1
        const wrapper = shallow(<ManageZones getZoneList={getZoneList} history={history} match={match} />)
        wrapper.setState({ fetchZonesList:zoneList.result})
        let addButton = wrapper.find("button.btn.btn-create");
        expect(addButton.length).toBe(1);
        addButton.simulate('click');
        expect(history.length).toEqual(expectedCount);
        expect(history.location.pathname).toEqual("/addOrEditZone/"+ match.params.plantId + "/" + match.params.pipelineId);
    });

    it("clicking back button trigger onClick event", async () => {
      let expectedCount = history.length + 1;
      const wrapper = shallow(<ManageZones getZoneList={getZoneList} history={history} match={match} />);
      let backButton = wrapper.find("span.cursor-pointer");
      expect(backButton.length).toBe(1);
      backButton.simulate('click');
      expect(history.length).toEqual(expectedCount);
      expect(history.location.pathname).toEqual("/pipeLineList/"+ match.params.plantId);
    });

    it("Function testing on mapDispatchToProps", () => {
        const mockParams = {
          params: {
            plantId: "plantId",
            pipelineId: "pipelineId",
            zoneId: "zoneId",
            multiple: "multiple"
          }
        };
        const dispatch = jest.fn();
        mapDispatchToProps(dispatch).getZoneList(
          mockParams.params.plantId,mockParams.params.pipelineId
        );
        expect(dispatch.mock.calls[0][0]).toEqual({
          type: GET_ZONE_LIST,
          plantId: "plantId",
          pipelineId: "pipelineId"
        });
        mapDispatchToProps(dispatch).zoneDeleteHandler(
            mockParams.params.plantId,mockParams.params.pipelineId,mockParams.params.multiple,mockParams.params.zoneId
        );
        expect(dispatch.mock.calls[1][0]).toEqual({
          type: ZONE_DELETE,
          plantId: "plantId",
          pipelineId: "pipelineId",
          zoneId: "zoneId",
          multiple: "multiple"
        });
    });
});