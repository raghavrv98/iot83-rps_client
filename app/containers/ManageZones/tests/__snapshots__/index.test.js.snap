// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manageAgentsSaga Saga match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      ManageZones
    </title>
    <meta
      content="Description of ManageZones"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p
        className="pd-l-30"
      >
        <span
          className="cursor-pointer"
          onClick={[Function]}
        >
          <button
            className="btn btn-transparent"
          >
            <i
              className="far fa-long-arrow-left"
            />
          </button>
          <FormattedMessage
            defaultMessage="Manage Pipelines"
            id="app.common.messages.managePipeline"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
        </span>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="Manage EHT Zone"
          id="app.common.messages.manageZone"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <span
          className="mr-l-15 mr-r-15"
        >
          |
        </span>
        <FormattedMessage
          defaultMessage="Plant Name"
          id="app.common.messages.plantName"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
         - 
        <span
          className="text-theme"
        />
        <span
          className="mr-l-15 mr-r-15"
        >
          |
        </span>
        <FormattedMessage
          defaultMessage="Pipeline Name"
          id="app.common.messages.pipelineName"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
         - 
        <span
          className="text-theme"
        />
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right align-items-center"
    />
  </div>
  <SkeletonLoader
    mode="fullView"
    skeleton="skeletonReactTable"
  />
</div>
`;
